amplify/src/main/java/com/github/stkent/amplify/tracking/AmplifyStateTracker.java

```diff
@@ -62,34 +75,21 @@ public static AmplifyStateTracker get(@NonNull final Context context) {
     }
 
     public AmplifyStateTracker setLogLevel(@NonNull final Logger.LogLevel logLevel) {
-        getLogger().setLogLevel(logLevel);
+        logger.setLogLevel(logLevel);
         return this;
     }
 
-    // TODO: when writing tests, loosen the visibility and override this method to provide an
-    // alternate implementation
-    private ISettings getSettings() {
-        if (settings == null) {
-            settings = Settings.getSharedInstance(applicationContext);
-        }
-
-        return settings;
-    }
-
-    // TODO: when writing tests, loosen the visibility and override this method to provide an
-    // alternate implementation
-    private ILogger getLogger() {
-        if (logger == null) {
-            logger = new Logger();
-        }
-
-        return logger;
-    }
-
     // constructors
 
-    private AmplifyStateTracker(@NonNull final Context context) {
+    private AmplifyStateTracker(
+            @NonNull final Context context,
+            @NonNull final ILogger logger) {
         this.applicationContext = context.getApplicationContext();
+        this.logger = logger;
+        this.lastEventTimePredicates = new LastTimeMap(logger, applicationContext);
```

Comment:
I mean you're instantiating the map inside the class, so this is no different from assigning it to the field directly. If you were intending to inject via the constructor for testing later, this won't help!